### PR TITLE
[docs] add EAGO to packages.toml

### PR DESCRIPTION
**Pull request by odow**
cc @DimitriAlston 

It'll take a while for the docs to run, but once they're finished this preview link will work:

https://jump.dev/JuMP.jl/previews/PR3561/packages/EAGO

We need to fix some typos, and add some more words to the allow-list (https://github.com/jump-dev/JuMP.jl/blob/master/docs/styles/Vocab/JuMP-Vocab/accept.txt).

The bigger problem is going to be the formatting and links. Documenter's Markdown is not the same as GitHub, so things will render differently. I'll take a look once the preview has built. But another alternative is to provide a simplified file just for the JuMP documentation.

This is what InfiniteOpt has done:

https://github.com/jump-dev/JuMP.jl/blob/c831cc6dad564b380b86c3bf0a74877f66d46773/docs/packages.toml#L141

https://github.com/infiniteopt/InfiniteOpt.jl/blob/master/docs/jump/README.md

### CODE REVIEW

**Comment by odow** [comment]
@DimitriAlston here are the vale failures. I've added some (not listed here) to the `accept.txt`.

[docs/src/packages/EAGO.md|20 col 317|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L20) [Google.Latin] Use 'for example' instead of 'e.g.,'.
[docs/src/packages/EAGO.md|106 col 3|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L106) [Google.DateFormat] Use 'July 31, 2016' format, not '6/15/2023'.
[docs/src/packages/EAGO.md|110 col 54|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L110) [Vale.Spelling] Did you really mean 'accomodate'?

Should be accommodate

[docs/src/packages/EAGO.md|123 col 146|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L123) [Google.Latin] Use 'for example' instead of 'e.g.,'.
[docs/src/packages/EAGO.md|123 col 166|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L123) [Vale.Spelling] Did you really mean 'Wergert'?

Should be Wengert

[docs/src/packages/EAGO.md|130 col 102|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L130) [Vale.Spelling] Did you really mean 'Github'?
[docs/src/packages/EAGO.md|134 col 70|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L134) [Vale.Spelling] Did you really mean 'Github'?

Should be GitHub

[docs/src/packages/EAGO.md|143 col 91|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L143) [Vale.Spelling] Did you really mean 'lexiographic'?

Should be lexicographic

[docs/src/packages/EAGO.md|155 col 17|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L155) [Vale.Spelling] Did you really mean 'bibtex'?

Should be BibTeX

[docs/src/packages/EAGO.md|172 col 213|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L172) [Vale.Spelling] Did you really mean 'contactors'?

Contractors?

[docs/src/packages/EAGO.md|178 col 134|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L178) [Google.EnDash] Use an em dash ('—') instead of '–'.
[docs/src/packages/EAGO.md|180 col 104|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L180) [Vale.Spelling] Did you really mean 'Optim'?

Optimization

[docs/src/packages/EAGO.md|180 col 119|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L180) [Vale.Spelling] Did you really mean 'Softw'?

Software

[docs/src/packages/EAGO.md|180 col 137|](https://github.com/jump-dev/JuMP.jl/blob/3cb3144a89a6da2b56def74843f8aa9f2db5303a/docs/src/packages/EAGO.md#L180) [Google.EnDash] Use an em dash ('—') instead of '–'.

**Comment by DimitriAlston** [comment]
I'm thinking it might be better if I do something similar to InfiniteOpt. EAGO's README from v0.8.1 has a lot of unnecessary stuff for this so a simplified version for getting started would be better. Also, I realized a lot of the typos I fixed are on my version and haven't been pushed to GitHub yet.

**Comment by odow** [comment]
No problem. Unfortunately, it's a bit of a hassle to test things. The best way is to push a commit to EAGO, then comment here with the hash, and I will rebuild the docs with it.

**Comment by odow** [comment]
I added `has_html`, so hopefully this time the images will render properly

**Comment by DimitriAlston** [comment]
The images probably won't work because I renamed and moved them at some point.
I'll have to make these edits tomorrow and send you updated info then.
Thanks for your help.

**Comment by DimitriAlston** [comment]
Alright, I have added a new README to EAGO for this at https://github.com/PSORLab/EAGO.jl/blob/master/docs/src/jump/README.md.
Am I able to edit this in the future, if necessary, without causing conflicts?

File: docs/packages.toml [hunk 1]
```diff
@@ -125,9 +125,10 @@
 [DAQP]
     user = "darnstrom"
     rev = "v0.5.1"
-# [EAGO]
-#     user = "PSORLab"
-#     rev = "v0.8.1"
+[EAGO]
+    user = "PSORLab"
+    rev = "v0.8.1"
```
Comment (odow) [inline docs/packages.toml:130]:
```suggestion
    rev = "75b09a9287a3264e2da1e003f0deb8278661c529"
```

**Comment by odow** [comment]
> Am I able to edit this in the future, if necessary, without causing conflicts?

Yeah. I've updated the commit hash. So if you change, we'll need to update it here. let's see how this one looks

File: docs/styles/Vocab/JuMP-Vocab/accept.txt [hunk 2]
```diff
@@ -231,6 +240,7 @@ Tillmann
 Ulungu
 Vandenberghe
 Vielma
+Wechsung
```
Comment (odow) [inline docs/styles/Vocab/JuMP-Vocab/accept.txt:243]:
```suggestion
Watson, H.A.J.
Wechsung
```

File: docs/packages.toml [hunk 3]
```diff
@@ -125,9 +125,11 @@
 [DAQP]
     user = "darnstrom"
     rev = "v0.5.1"
-# [EAGO]
-#     user = "PSORLab"
-#     rev = "v0.8.1"
+[EAGO]
+    user = "PSORLab"
+    rev = "75b09a9287a3264e2da1e003f0deb8278661c529"
```
Comment (odow) [inline docs/packages.toml:130]:
```suggestion
    rev = "bea34bd61d686721b25dd2a68fbe64325a56a28e"
```

**Comment by DimitriAlston** [comment]
Oh, I think we forgot to remove `has_html = true`.

File: docs/packages.toml [hunk 4]
```diff
@@ -125,9 +125,11 @@
...
     user = "darnstrom"
     rev = "v0.5.1"
-# [EAGO]
-#     user = "PSORLab"
-#     rev = "v0.8.1"
+[EAGO]
+    user = "PSORLab"
+    rev = "bea34bd61d686721b25dd2a68fbe64325a56a28e"
+    filename = "docs/src/jump/README.md"
+    has_html = true
```
Comment (odow) [inline docs/packages.toml:132]:
```suggestion
```

**Comment by odow** [comment]
My bad. See you in another hour.

File: docs/packages.toml [hunk 5]
```diff
@@ -125,9 +125,10 @@
 [DAQP]
     user = "darnstrom"
     rev = "v0.5.1"
-# [EAGO]
-#     user = "PSORLab"
-#     rev = "v0.8.1"
+[EAGO]
+    user = "PSORLab"
+    rev = "bea34bd61d686721b25dd2a68fbe64325a56a28e"
```
Comment (odow) [inline docs/packages.toml:130]:
```suggestion
    rev = "8bba4cf6d9ef1602b4cfe3ea31b847f21ce25837"
```

**Comment by odow** [comment]
Thanks for your patience @DimitriAlston

**Comment by DimitriAlston** [comment]
Thanks for your help!

**Pull request merged**